### PR TITLE
Add filter to sale report, register sale datetime, register beneficiaries  - refs #7768

### PR DESCRIPTION
**Pull request by AngelFQC**

### CODE REVIEW

File: plugin/buycourses/lang/english.php [hunk 1]
```diff
@@ -84,3 +84,7 @@
 $strings['InfoApiStepTwo'] = "In paragraph <strong>API access</strong>, click <strong>Update</strong>";
 $strings['InfoApiStepThree'] = "In Option 2 (Request API credentials to create your own API username and password), click the <strong>View API Signature</strong> link, and copy the credentials shown right into the BuyCourses plugin form";
 $strings['ErrorOccurred'] = "<strong>An error ocurred</strong>. Code: %s. Message: %s. Please contact to platform admin";
+$strings['VisibleInCatalog'] = "Visible in catalog";
+$strings['Beneficiaries'] = "Beneficiaries";
+$strings['AvailableCourse'] = "Curso disponible";
```
Comment (ywarnier) [inline plugin/buycourses/lang/english.php]: Still in Spanish. Should be "Available courses"

File: plugin/buycourses/lang/french.php [hunk 2]
```diff
@@ -84,3 +84,7 @@
 $strings['InfoApiStepTwo'] = "Dans la section <strong>API d\'accès</strong>, cloquer sur l\'option <strong>Mettre à jour</strong>";
 $strings['InfoApiStepThree'] = "Dans l\'option 2 de Configuration des données et permissions API, cliquer sur <strong>Voir signature API</strong>. Copier ces donées dans le formulaire de configuration de ce plugin";
 $strings['ErrorOccurred'] = "<strong>Une erreur est survenue</strong>. Code: %s. Message: %s. Veuillez contacter l'administrateur de la plateforme";
+$strings['VisibleInCatalog'] = "Visible dans le catalogue";
+$strings['Beneficiaries'] = "Bénéficiaires";
+$strings['AvailableCourse'] = "Cours disponibles";
+$strings['ShowOnCourseCatalog'] = "Afficher sur le catalogue des cours";
```
Comment (ywarnier) [inline plugin/buycourses/lang/french.php]: Afficher dans le catalogue de cours

Comment (AngelFQC) [inline plugin/buycourses/lang/french.php]: Fixed

File: plugin/buycourses/lang/english.php (hunk 1 above)
Comment (AngelFQC) [inline plugin/buycourses/lang/english.php]: Fixed

**Pull request merged**